jwst/pipeline/calwebb_spec2.py

```diff
@@ -292,12 +290,14 @@ def process_exposure_product(
 
             # Call the resample_spec step for 2D slit data
             resampled = calibrated.copy()
+            resampled = self.pixel_replace(resampled)
```

Comment:
Instead of adding the call to pixel_replace in several places within calwebb_spec2, would it be cleaner/easier to just add it to the list of steps that are called by `_process_nirspec_slits` and `_process_common`, or do you not want it to be applied to the `calibrated` data model, so that the "cal" file does not have it applied?

Reply:
I really was not sure about what other data would fall in the final else statement that did not call pixel_replace. 
So I just added calling pixel_replace for each different kind of data that I know we want to run the step.